main/src/cgeo/geocaching/maps/CGeoMap.java

```diff
@@ -1316,28 +1300,22 @@ private void doDisplayRun() {
 
             // display caches
             final List<Geocache> cachesToDisplay = caches.getAsList();
-            final List<Waypoint> waypointsToDisplay = new ArrayList<>(waypoints);
```

Comment:
Copying the waypoints does not solve the problem. IIRC we already had this discussion.